internal/controller/storageclient_controller.go

```diff
@@ -421,9 +423,51 @@ func (r *storageClientReconcile) reconcilePhases() (ctrl.Result, error) {
 		}
 	}
 
+	//Once the owner references are removed from resources, StorageClaim is deleted
+	if err = r.deleteStorageClaims(); err != nil {
+		return reconcile.Result{}, err
+	}
 	return reconcile.Result{}, nil
 }
 
+func (r *storageClientReconcile) deleteStorageClaims() error {
+	storageClaimFinalizer := "storageclaim.ocs.openshift.io"
+	expectedStorageClaimNames := []string{
+		fmt.Sprintf("%s-ceph-rbd", r.storageClient.Name),
+		fmt.Sprintf("%s-cephfs", r.storageClient.Name),
+	}
+
+	for _, claimName := range expectedStorageClaimNames {
+		storageClaim := &metav1.PartialObjectMetadata{}
+		storageClaim.SetGroupVersionKind(schema.GroupVersionKind{
+			Group:   "ocs.openshift.io",
+			Version: "v1alpha1",
+			Kind:    "StorageClaim",
+		})
+		storageClaim.Name = claimName
+
+		if err := r.Client.Get(r.ctx, types.NamespacedName{Name: claimName}, storageClaim); err != nil {
```

Comment:
The get should go after the delete. If there was no finalizer the get might just return a NotFound and we don't need to do anything else after that point